src/NUnitFramework/tests/Internal/TestExecutionContextTests.cs

```diff
@@ -475,6 +475,14 @@ public async Task AsyncTestCanAccessOutWriter()
 
 #endregion
 
+#region SendMessage
+        [Test]
+        public void CanSendMessage()
+        {
+            TestExecutionContext.CurrentContext.SendMessage("destination", "message");
```

Comment:
This test merely verifies that the call does not throw. It doesn't indicate what happens as a result. I believe you need to follow a strategy similar to `TextOutputTests`, where the class itself implements a listener and verifies that it receives the proper message. That test would work within it's own, specially created context, in order to avoid polluting the actual test run with a fake message to "destination". 

This is probably better done in `SendMessageTests` rather than here.